client/src/App.tsx

```diff
@@ -1,41 +1,41 @@
 import React, { FunctionComponent, Suspense, useEffect, useCallback } from 'react';
 import { Route, Routes } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
 
-// API
-import { useAuthorizeMutation, useAuthenticateMutation } from './services/authApi';
+// Store
+import { authorize, authenticate } from 'redux/actionCreators/auth';
 
 // Types
-import { AuthorizationRequest, AuthenticationResponse } from './types';
+import type { AuthorizationRequest } from 'types';
 
 // Components
-import Layout from './common/hoc/Layout';
+import Layout from 'hoc/Layout/Layout';
 
 // Lazy components
-const Home = React.lazy(() => import('./common/pages/Home'));
-const About = React.lazy(() => import('./common/pages/About'));
+const HomePage = React.lazy(() => import('components/HomePage/HomePage'));
+const AboutPage = React.lazy(() => import('components/AboutPage/AboutPage'));
 
 // Routes
 const routes = [
-  { path: '/', name: 'Home', Component: Home },
-  { path: '/about', name: 'About', Component: About }
+  { path: '/', name: 'Home', Component: HomePage },
+  { path: '/about', name: 'About', Component: AboutPage }
 ];
 
 const App: FunctionComponent = () => {
-  const [authorize] = useAuthorizeMutation();
-  const [authenticate] = useAuthenticateMutation();
+  const dispatch = useDispatch();
 
   const handleAuth = useCallback(async () => {
-    const { token }: AuthenticationResponse = await authenticate().unwrap();
+    const token = await dispatch(authenticate());
 
     const credentials: AuthorizationRequest = {
       user: process.env.USER,
       password: process.env.PASSWORD
     };
 
     if (!token) {
-      await authorize(credentials);
+      await dispatch(authorize(credentials));
     }
-  }, [authenticate, authorize]);
+  }, [dispatch]);
```

Comment:
Why you use useCallback and 'dispatch' in dependencies array here? 
You are not passing a callback using dispatch to a child component. 
'react-redux' documentation says that "The dispatch function reference will be stable as long as the same store instance is being passed to the <Provider>. Normally, that store instance never changes in an application."
I think 'handleAuth' will not change for all life app because App component will not update. 
I think you don't should to use useCallback and call handleAuth in useEffect witch empty dependencies array

Reply:
ESLint wants it to be in such a way.

Reply:
I think you can create handleAuth function without useCallback and call this function in useEffect with empty dependencies array. Then you get token, when app started. 
If I'm right, your App component doesn't update for all application life. That's why your handleAuth function will not create again and again. Consequently you don't need a hook useCallback.
But! I could be wrong. I think Julia can help find truth in our discussion :)

Reply:
@ViliusGaizutis please find the answer and explain Andrew why `dispatch` needs to be mentioned in dependency array exactly in your case and in your project. Answer “eslint shows error otherwise” is not a correct developer answer, you need to understand why it works like this, you are a code owner :)

Reply:
React Redux docs say: "The dispatch function reference will be stable as long as the same store instance is being passed to the <Provider>. Normally, that store instance never changes in an application.
However, the React hooks lint rules do not know that dispatch should be stable, and will warn that the dispatch variable should be added to dependency arrays for useEffect and useCallback. The simplest solution is to do just that."
(https://react-redux.js.org/api/hooks#usedispatch)